### PR TITLE
Add shellcode to CobaltStrike carver (#10)

### PR DESCRIPTION
**Pull request by mgreen27**
Added shellcode parsing into Cobalt Strike carver.   
- will check for shellcode stager if no beacon is found.  
- dedup hits and maintain performance.  

image

### CODE REVIEW

File: artifacts/definitions/Windows/Carving/CobaltStrike.yaml [hunk 1]
```diff
@@ -1,44 +1,43 @@
...
-
+  
 parameters:
   - name: TargetBytes
     default:
   - name: TargetFileGlob
     default:
   - name: PidRegex
     default: .
-    type: regex
```
Comment (scudette) [inline artifacts/definitions/Windows/Carving/CobaltStrike.yaml:38]: Please keep the regex types.

File: artifacts/definitions/Windows/Carving/CobaltStrike.yaml [hunk 2]
```diff
@@ -116,23 +127,23 @@ export: |
...
-
+        
         # 0x000d:HttpPostHeader,0x0003:Type, adding length check as we can not rely on termination
         ["__http_post_header_prefix", 0, "String",{"term_hex": "000d0003", length: 10000, max_length: 10000}],
         ["__HttpPostHeader_length","x=>len(list=x.__http_post_header_prefix) + 4","uint16b"],
         ["__HttpPostHeader","x=>len(list=x.__http_post_header_prefix) + 6","String",{"term":"***NOTERM***", "length": "x=> x.__HttpPostHeader_length"}],
         ["HttpPostHeader",0,"Value",{ "value": "x=>format(format='%s', args=[regex_replace(source=x.__HttpPostHeader, re='[^ -~\\r\\n]', replace='')])" }],
         #["HttpPostHeader",0,"Value",{ "value": "x=>'base64:' + base64encode(string=x.__HttpPostHeader)" }], #uncomment to return base64 encoded raw HttpPostHeader
-
+
```
Comment (scudette) [inline artifacts/definitions/Windows/Carving/CobaltStrike.yaml:146]: Please enable trailing white space removal

File: artifacts/definitions/Windows/Carving/CobaltStrike.yaml (hunk 1 above)
Comment (mgreen27) [reply artifacts/definitions/Windows/Carving/CobaltStrike.yaml:38]: added in latest commit

File: artifacts/definitions/Windows/Carving/CobaltStrike.yaml (hunk 2 above)
Comment (mgreen27) [reply artifacts/definitions/Windows/Carving/CobaltStrike.yaml:145]: this should be updated now.

**Pull request merged**